### ISSUE TITLE
on_touch is not working

### ISSUE THREAD

**Issue by hugoerg56** [HipByte/motion-game#14]
Ruby 2.2.0
Motion-game (1.1)
Rubymotion 4.9

After run the app:

```
*** Starting simulator
2016-02-24 14:05:48.467 urbanz[11864:111669] Error loading /System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib:  dlopen(/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib, 262): no suitable image found.  Did find:
	/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib: mach-o, but not built for iOS simulator
2016-02-24 14:05:48.468 urbanz[11864:111669] Cannot find function pointer IOHIDLibFactory for factory 13AA9C44-6F1B-11D4-907C-0005028F18D5 in CFBundle/CFPlugIn 0x10f27c950 </System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin> (bundle, not loaded)
2016-02-24 14:05:48.468 urbanz[11864:111669] Error loading /System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib:  dlopen(/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib, 262): no suitable image found.  Did find:
	/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib: mach-o, but not built for iOS simulator
2016-02-24 14:05:48.468 urbanz[11864:111669] Cannot find function pointer IOHIDLibFactory for factory 13AA9C44-6F1B-11D4-907C-0005028F18D5 in CFBundle/CFPlugIn 0x10f27c950 </System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin> (bundle, not loaded)
2016-02-24 14:05:48.468 urbanz[11864:111669] Error loading /System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib:  dlopen(/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib, 262): no suitable image found.  Did find:
	/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib: mach-o, but not built for iOS simulator
2016-02-24 14:05:48.469 urbanz[11864:111669] Cannot find function pointer IOHIDLibFactory for factory 13AA9C44-6F1B-11D4-907C-0005028F18D5 in CFBundle/CFPlugIn 0x10f27c950 </System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin> (bundle, not loaded)
2016-02-24 14:05:48.469 urbanz[11864:111669] Error loading /System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib:  dlopen(/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib, 262): no suitable image found.  Did find:
	/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib: mach-o, but not built for iOS simulator
2016-02-24 14:05:48.469 urbanz[11864:111669] Cannot find function pointer IOHIDLibFactory for factory 13AA9C44-6F1B-11D4-907C-0005028F18D5 in CFBundle/CFPlugIn 0x10f27c950 </System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin> (bundle, not loaded)
2016-02-24 14:05:48.469 urbanz[11864:111669] Error loading /System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib:  dlopen(/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib, 262): no suitable image found.  Did find:
	/System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib: mach-o, but not built for iOS simulator
2016-02-24 14:05:48.469 urbanz[11864:111669] Cannot find function pointer IOHIDLibFactory for factory 13AA9C44-6F1B-11D4-907C-0005028F18D5 in CFBundle/CFPlugIn 0x10f27c950 </System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin> (bundle, not loaded)

```

The app don't crash, still working. but the buttons on_touch event dont't work:

```
def add_menu_option
  button = MG::Button.new("New Game")
  button.font_size = 35
  button.position = [MG::Director.shared.size.width / 2, (MG::Director.shared.size.height - 200)]
  button.on_touch { MG::Director.shared.replace(IntroScene.new) }

  add button
end
```

on this button the click event looks good visually but is not opening the IntroScene and don't get anything on terminal, is not working at all the on_touch event. any idea @Watson1978 @jjaffeux ?

**Comment by Watson1978** [comment]
Actually, I think this warning message introduced by cocos2d-x.
http://discuss.cocos2d-x.org/t/switching-to-xcode-6-output-errors/17089

If this message is a concern, please report to cocos2d-x.

**Comment by hugoerg56** [comment]
Ok I ignore the warnings, but button.on_touch { MG::Director.shared.replace(IntroScene.new) } is not called at all. Any idea?

**Comment by Watson1978** [comment]
If you search issue at cocos2d-x forum or issue tracker, you might find a workaround. 
J have no idea to solve your issue now.

**Comment by hugoerg56** [comment]
@Watson1978 Sorry to bother much but I really like to use motion-game, the only solution I saw in many places was: 

You can disable them by setting "Linker Behavior" in "Build Settings" to "Link SDK Assemblies Only".

but I have no idea how to do this or if posible in rubymotion?

**Comment by Watson1978** [comment]
OK, if you use motion-game v1.1, actually Button#on_touch does not work as expected.
However, if you build motion-game library by yourself,
```
$ git clone https://github.com/HipByte/motion-game.git
$ cd motion-game.git
$ rake build:ios
```
then Button#on_touch will work.

@lrz motion-game v1.1 is missing something?

**Comment by Watson1978** [comment]
This issue should be fixed with motion-game v1.1.1.
